tyr-runner/src/main/java/org/jboss/tyr/ci/TeamCityCI.java

```diff
@@ -101,29 +124,100 @@ public void triggerBuild(JsonObject prPayload) {
         }
     }
 
-
-    private String encryptCredentials(String username, String password) {
-        String authStr = username + ":" + password;
-        return DatatypeConverter.printBase64Binary(authStr.getBytes(StandardCharsets.UTF_8));
+    @Override
+    public void triggerRetestingFailed(JsonObject prPayload) {
+        setTriggerBuildProperties(prPayload);
+        if (!branchMappings.containsKey(branch)) {
+            return;
+        }
+        setSnapshotDependencies(buildId, true);
```

Comment:
Can you refactor this somehow please? This method and triggerRetesting differ only in the boolean value passed to this invocation.

Reply:
Yes, of course. :)